test/bntests.pl

```diff
@@ -0,0 +1,157 @@
+#! /usr/bin/env perl
+# Copyright 2008-2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+# Run the tests specified in bntests.txt, as a check against OpenSSL.
+use strict;
+use warnings;
+use Math::BigInt;
+
+sub bn()
+{
+    my $x = shift;
+    my ($sign, $hex) = ($x =~ /^([+\-]?)(.*)$/);
+
+    $hex = '0x' . $hex if $hex !~ /^0x/;
+    return Math::BigInt->from_hex($sign.$hex);
+}
+
+sub evaluate()
+{
+    my ( $lineno ) = shift;
+    my ( %s ) = @_;
+
+    if ( defined $s{'Sum'} ) {
+        # Sum = A + B
+        my $sum = &bn($s{'Sum'});
```

Comment:
Function calls do not need to be preceded with `&` any more.  That's perl 4... or was it even perl 3?